hangul/hangulctype.c

```diff
@@ -73,6 +85,44 @@ hangul_is_choseong(ucschar c)
 ;
 }
 
+
+void ucs4_to_utf8(char *buf, const ucschar *ucs4, size_t bufsize)
+{
+    size_t n;
+    char*  inbuf;
+    size_t inbytesleft;
+    char*  outbuf;
+    size_t outbytesleft;
+    size_t ret;
+    iconv_t cd;
+
+    for (n = 0; ucs4[n] != 0; n++)
+	continue;
+
+    if (n == 0) {
+	buf[0] = '\0';
+	return;
+    }
+
+    cd = iconv_open("UTF-8", UCS4);
+    if (cd == (iconv_t)(-1))
+	return;
+
+    inbuf = (char*)ucs4;
+    inbytesleft = n * 4;
+    outbuf = buf;
+    outbytesleft = bufsize;
+    ret = iconv(cd, &inbuf, &inbytesleft, &outbuf, &outbytesleft);
+    (void)ret;
+
+    iconv_close(cd);
+
+    if (outbytesleft > 0)
+	*outbuf = '\0';
+    else
+	buf[bufsize - 1] = '\0';
```

Comment:
I see that you've used the same formatting as the rest of the code but I would like to note that I'm not a fan of it and would much prefer if the new functions follow a more sane style guide.
If the rest of the team are ok with this I can accept it.

Reply:
I copied this function from the sample application [here](https://github.com/abaltatech/libhangul/blob/main/test/hangul.c). That is why I haven't changed the style in it.